Dockerfile

```diff
@@ -83,23 +87,29 @@ RUN pip3 install \
     hidapi \
     noiseprotocol \
     protobuf \
+    mypy-protobuf \
     ecdsa \
     tzlocal \
     semver
 
 # Python modules for CI
-RUN pip3 install \
+RUN python3 -m pip install \
     pylint \
     pylint-protobuf \
     black \
-    mypy==0.720 \
-    mypy-protobuf==1.13
+    mypy
```

Comment:
why drop the version? ideally every software that Docker gets is pinned, otherwise a simple rebuild could break CI (happened frequently in the BitBox App until I pinned the relevant packages).

Reply:
shouldn't have versions here, they should be in `requirements.txt`

Reply:
oh, or if packages only needed for CI it should take the latest and break so that we don't get stuck on old versions. This won't break the build system because it's only related to linting.

Reply:
They are in requirements.txt and ideally we'd be installing from that file in the Dockerfile.

Still disagree. Rebuilding a Docker image should not cause any breakage if we can help it, and not pinning versions will cause breaks.

Not getting stuck on old versions: you want to upgrade them at your own pace and controlled, and not deal with it when it hits you at an inconvenient time.

Trust me, it was very inconvenient in the BitBox App until I fixed the versions.

Reply:
To elaborate: similar to the lazy linting approach, I very much dislike having to deal with unrelated issues when I am working on something.

Reply:
Yeah I see your point and added hard coded versions. Like mentioned below we are unfortunately using incompatible versions right now. The error doesn't seem fatal though, so my suggestion is that we live with it.

Reply:
I also get that you don't prefer lazy linting and want to move the whole project at once due to the reasons you stated (we will have to agree to disagree on that one). I wouldn't call it lazy that CI only looks at modified files though. It is however lazy to not check and fix all the files at the time new checks are introduced. But to me those are two unrelated things.

Reply:
I don't mind not moving the whole project at once, file by file or similar is fine, as long as it doesn't force me to do it when I don't expect it :smile: 

Thanks!